tstate/tstate_view.go

```diff
@@ -162,8 +162,17 @@ func (ts *TStateView) KeyOperations() (map[string]uint16, map[string]uint16) {
 }
 
 // checkScope returns whether [k] is in scope and has appropriate permissions.
-func (ts *TStateView) checkScope(_ context.Context, k []byte, perm state.Permissions) bool {
-	return ts.scope[string(k)].Has(perm)
+func (ts *TStateView) checkScope(_ context.Context, k []byte, perms ...state.Permissions) bool {
+	// At most only 2 key permissions are passed that we need to check. The case of
+	// 1 permission may be a single key permission or a combination (i.e., Allocate|Write).
+	switch len(perms) {
```

Comment:
Suggested way to use bit ops for this below...even if we did keep this, I would recommend iterating through a for loop here instead of writing through these cases.